clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/common/AzureUtilities.groovy

```diff
@@ -112,6 +113,30 @@ class AzureUtilities {
     resourceId.split(PATH_SEPARATOR).last()
   }
 
+  // id = "/subscriptions/***-***-***/resourceGroups/***/providers/Microsoft.Network/networkInterfaces/nic1/ipConfigurations/ipconfig1"
+  static String getResourceNameFromId(String id) {
```

Comment:
Different Azure resource provider. You have this hard-coded to handle only Microsoft.Network resources. What if we need to get the name for a Microsoft.Compute resource?